src/Publisher/Publisher.php

```diff
@@ -42,4 +43,12 @@ public final function unpublish(string $key): SuccessResult
      * @throws StreamxClientException If command filed.
      */
     public abstract function send(Message $message): SuccessResult;
+
+    /**
+     * Sends the provided ingestion messages to the Ingestion endpoint.
+     * @param Message[] $messages Ingestion messages.
+     * @return MessageStatus[] with SuccessResult and/or FailureResponse from ingestion endpoint for each input message (in order).
+     * @throws StreamxClientException If a critical error occurred.
+     */
+    public abstract function sendMulti(array $messages): array;
```

Comment:
Unfortunately, not in php..
This would raise compilation error: "method with same name already exists"

Reply:
Maybe even in java client we could stick to different names of the two methods - since the new method has a different behavior - it doesn't throw exception in failure, it returns failures in response.